runtime/compiler/CMakeLists.txt

```diff
@@ -395,7 +395,7 @@ set_source_files_properties(
 
 if(OMR_OS_LINUX)
 	set_property(TARGET j9jit APPEND_STRING PROPERTY
-		LINK_FLAGS " -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/build/scripts/j9jit.linux.exp")
+		LINK_FLAGS " -Wl,-z,noexecstack -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/build/scripts/j9jit.linux.exp")
```

Comment:
If a single assembler were used on each platform, this should not be necessary (see https://github.com/eclipse-openj9/openj9/issues/20956#issuecomment-2666741850).

This seems a reasonable, perhaps short-term, solution.